Describe recommendations for destroy method [HZ-4563] (#1057)
hazelcast/hz-docs: docs/modules/data-structures/pages/distributed-data-structures.adoc

Therefore, keep in mind that a
data structure is recreated when you perform an operation on it even after you have destroyed it.

> NOTE: If you plan to use the distributed object later, prefer clearing the object using `clear()` method or equivalent instead of destroying it.

Reviewer: ```suggestion
NOTE: If you plan to use the distributed object later, clear the object using the `clear()` method or equivalent instead of destroying it.
```

> WARNING: Do not use distributed object concurrently with `destroy()` invocation. This may make the object not fully usable.

Reviewer: ```suggestion
WARNING: Do not use the distributed object concurrently with the `destroy()` invocation, as this can result in the object not being fully usable.
```

> WARNING: Do not use the distributed object concurrently with the `destroy()` invocation, as this can result in the object not being fully usable.

Reviewer 2: > I think unused stale references shouldn't cause problems except memory consumption on the client.

that's true

Therefore, keep in mind that a
data structure is recreated when you perform an operation on it even after you have destroyed it.

NOTE: If you plan to use the distributed object later, clear the object using the `clear()` method or equivalent instead of destroying it.

WARNING: Do not use the distributed object concurrently with the `destroy()` invocation, as this can result in the object not being fully usable.

> WARNING: If you want to use the destroyed distributed object again, make sure that you do not hold any old references, but obtain a new, fresh reference from `HazelcastInstance.getXXX` method, 

Reviewer: ```suggestion
WARNING: If you want to use the destroyed distributed object again, do not hold any old references; instead, obtain a new reference using the `HazelcastInstance.getXXX` method, 
```

> otherwise the object may not be fully usable.

Reviewer: ```suggestion
otherwise the object might not be fully usable.
```
